src/Nethermind/Nethermind.Core.Test/Blockchain/TestBlockchain.cs

```diff
@@ -102,24 +95,20 @@ protected TestBlockchain()
     public static readonly DateTime InitialTimestamp = new(2020, 2, 15, 12, 50, 30, DateTimeKind.Utc);
 
     public static readonly UInt256 InitialValue = 1000.Ether();
-    public IHeaderValidator HeaderValidator => _fromContainer.HeaderValidator;
 
     protected AutoCancelTokenSource _cts;
     public CancellationToken CancellationToken => _cts.Token;
 
     private TestBlockchainUtil TestUtil => _fromContainer.TestBlockchainUtil.Value;
     private PoWTestBlockchainUtil PoWTestUtil => _fromContainer.PoWTestBlockchainUtil.Value;
-
-    public IBlockValidator BlockValidator => _fromContainer.BlockValidator;
-
-    public IManualBlockProductionTrigger BlockProductionTrigger => _fromContainer.BlockProductionTrigger;
+    private IManualBlockProductionTrigger BlockProductionTrigger => _fromContainer.BlockProductionTrigger;
 
     public ManualTimestamper Timestamper => _fromContainer.ManualTimestamper;
-    public IBlocksConfig BlocksConfig => Container.Resolve<IBlocksConfig>();
+    protected IBlocksConfig BlocksConfig => Container.Resolve<IBlocksConfig>();
 
     public ProducedBlockSuggester Suggester { get; protected set; } = null!;
 
-    public IExecutionRequestsProcessor MainExecutionRequestsProcessor => ((MainBlockProcessingContext)_fromContainer.MainProcessingContext).LifetimeScope.Resolve<IExecutionRequestsProcessor>();
+    public IExecutionRequestsProcessor MainExecutionRequestsProcessor => ((AutoMainProcessingContext)_fromContainer.MainProcessingContext).LifetimeScope.Resolve<IExecutionRequestsProcessor>();
```

Comment:
why `Auto` and not `Main`?

Reply:
There is already `MainProcessingContext` in prod. Its manual and not auto.